pkg/controller/poolcoordinator/podbinding/podbinding_controller.go

```diff
@@ -0,0 +1,273 @@
+/*
+Copyright 2023 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podbinding
+
+import (
+	"context"
+	"flag"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	appconfig "github.com/openyurtio/openyurt/cmd/yurt-manager/app/config"
+	"github.com/openyurtio/openyurt/pkg/controller/poolcoordinator/constant"
+	"github.com/openyurtio/openyurt/pkg/projectinfo"
+)
+
+func init() {
+	flag.IntVar(&concurrentReconciles, "podbinding-controller", concurrentReconciles, "Max concurrent workers for podbinding-controller controller.")
+}
+
+type Subject int32
+
+const (
+	NODE Subject = 0
+	POD  Subject = 1
+)
+
+const (
+	controllerName = "podbinding-controller"
+)
+
+var (
+	concurrentReconciles = 5
+
+	notReadyToleration = corev1.Toleration{
+		Key:      corev1.TaintNodeNotReady,
+		Operator: corev1.TolerationOpExists,
+		Effect:   corev1.TaintEffectNoExecute,
+	}
+
+	unreachableToleration = corev1.Toleration{
+		Key:      corev1.TaintNodeUnreachable,
+		Operator: corev1.TolerationOpExists,
+		Effect:   corev1.TaintEffectNoExecute,
+	}
+	defaultTolerationSeconds = 300
+)
+
+type ReconcilePodBinding struct {
+	client.Client
+	podBindingClient kubernetes.Interface
+	role             Subject // Node or Pod
+}
+
+// Add creates a PodBingding controllers and adds them to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(_ *appconfig.CompletedConfig, mgr manager.Manager) error {
+	r := newReconciler(mgr, NODE)
+	c, err := controller.New(controllerName+"-node", mgr, controller.Options{
+		Reconciler: r, MaxConcurrentReconciles: concurrentReconciles,
+	})
+	if err != nil {
+		return err
+	}
+	err = c.Watch(&source.Kind{Type: &corev1.Node{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	r = newReconciler(mgr, POD)
+	c, err = controller.New(controllerName+"-pod", mgr, controller.Options{
```

Comment:
as we discussed in the meeting, it is very complicated and confused for users to trigger this feature in two different ways(add annotation on node or add annotation on pod). on the other hand, it is very hard to add annotation to pod based on the workload like deployment, statefulset, and so on.

so we will only keep the feature that add annotation on node to trigger pod binding feature, and remove add annotation on pod mode.